### PR TITLE
Custom cmdline flag

### PR DESCRIPTION
**Pull request by florin0x01**

### CODE REVIEW

File: obs-browser-plugin.cpp [hunk 1]
```diff
@@ -356,23 +372,27 @@ static void BrowserShutdown(void)
...
 {
 	if (!os_atomic_set_bool(&manager_initialized, true)) {
-		blog(LOG_INFO, "Initialize obs_browser");
 #ifdef USE_UI_LOOP
-		BrowserInit();
+		blog(LOG_INFO, "obs_browser_initialize, using UI_LOOP, call BrowserInit");
+		BrowserInit(settings, source);
 #else
-		manager_thread = thread(BrowserManagerThread);
+		 blog(LOG_INFO, "obs_browser_initialize, NOT using UI_LOOP");
```
Comment (EddyGharbi) [inline obs-browser-plugin.cpp]: extra space here

File: obs-browser-plugin.cpp [hunk 2]
```diff
@@ -221,12 +235,12 @@ static obs_properties_t *browser_source_get_properties(void *data)
 	return props;
 }
 
-static CefRefPtr<BrowserApp> app;
 
-static void BrowserInit(void)
+static void BrowserInit(obs_data_t *settings_obs, obs_source_t *source)
```
Comment (EddyGharbi) [inline obs-browser-plugin.cpp]: Why is the source being passed? It seems like we only need the settings no?

File: obs-browser-plugin.cpp [hunk 3]
```diff
@@ -392,17 +412,34 @@ void RegisterBrowserSource()
...
+        blog(LOG_INFO, "Browserapp pointer: %p", app.get());
+        
+        return bs;
 	};
 	info.destroy = [](void *data) {
 		delete static_cast<BrowserSource *>(data);
 	};
 	info.update = [](void *data, obs_data_t *settings) {
-		static_cast<BrowserSource *>(data)->Update(settings);
+        BrowserSource *bs = static_cast<BrowserSource *>(data);
```
Comment (EddyGharbi) [inline obs-browser-plugin.cpp]: styling issues

File: obs-browser-plugin.cpp [hunk 4]
```diff
@@ -648,7 +685,7 @@ bool obs_module_load(void)
 #if defined(__APPLE__) && CHROME_VERSION_BUILD < 4183
 	// Make sure CEF malloc hijacking happens early in the process
 	if(isHighThanBigSur())
-		obs_browser_initialize();
+		obs_browser_initialize(nullptr, nullptr);
```
Comment (EddyGharbi) [inline obs-browser-plugin.cpp:683]: Is there a specific reason why we shouldn't pass the data in that case?

**Comment by EddyGharbi** [comment]
General comment: this PR adds too many logs, please reduce them.

File: obs-browser-plugin.cpp (hunk 4 above)
Comment (florin0x01) [reply obs-browser-plugin.cpp:683]: we don't have the data here.

File: obs-browser-plugin.cpp (hunk 3 above)
Comment (florin0x01) [reply obs-browser-plugin.cpp]: done

File: obs-browser-plugin.cpp (hunk 2 above)
Comment (florin0x01) [reply obs-browser-plugin.cpp]: indeed

File: obs-browser-plugin.cpp (hunk 1 above)
Comment (florin0x01) [reply obs-browser-plugin.cpp]: done

**Review by EddyGharbi — APPROVED** [review]

**Pull request merged**